x-pack/plugin/stack/src/javaRestTest/java/org/elasticsearch/xpack/stack/EcsDynamicTemplatesIT.java

```diff
@@ -0,0 +1,416 @@
+        assertNotNull(errorMessage, mappings);
+        assertThat(errorMessage, mappings, instanceOf(Map.class));
+        Object dynamicTemplates = ((Map<?, ?>) mappings).get("mappings");
+        assertNotNull(errorMessage, dynamicTemplates);
+        assertThat(errorMessage, dynamicTemplates, instanceOf(Map.class));
+        assertEquals(errorMessage, 1, ((Map<?, ?>) dynamicTemplates).size());
+        assertTrue(errorMessage, ((Map<?, ?>) dynamicTemplates).containsKey("dynamic_templates"));
+        ecsDynamicTemplates = (Map<String, Object>) dynamicTemplates;
+    }
+
+    @SuppressForbidden(reason = "Opening socket connection to read ECS definitions from ECS GitHub repo")
+    private static void prepareEcsDefinitions() throws IOException {
+        Map<String, ?> ecsFlatFieldsRawMap;
+        URL ecsDefinitionsFlatFileUrl = new URL(ECS_FLAT_FILE_URL);
+        try (InputStream ecsDynamicTemplatesIS = ecsDefinitionsFlatFileUrl.openStream()) {
+            try (
+                XContentParser parser = XContentFactory.xContent(XContentType.YAML)
+                    .createParser(XContentParserConfiguration.EMPTY, ecsDynamicTemplatesIS)
+            ) {
+                ecsFlatFieldsRawMap = parser.map();
+            }
+        }
+        String errorMessage = String.format(
+            Locale.ENGLISH,
+            "ECS flat mapping file at %s has changed, this test needs to be adjusted",
+            ECS_FLAT_FILE_URL
+        );
+        assertFalse(errorMessage, ecsFlatFieldsRawMap.isEmpty());
+        Map.Entry<String, ?> fieldEntry = ecsFlatFieldsRawMap.entrySet().iterator().next();
+        assertThat(errorMessage, fieldEntry.getValue(), instanceOf(Map.class));
+        Map<?, ?> fieldProperties = (Map<?, ?>) fieldEntry.getValue();
+        assertFalse(errorMessage, fieldProperties.isEmpty());
+        Map.Entry<?, ?> fieldProperty = fieldProperties.entrySet().iterator().next();
+        assertThat(errorMessage, fieldProperty.getKey(), instanceOf(String.class));
+
+        OMIT_FIELDS.forEach(ecsFlatFieldsRawMap::remove);
+
+        // noinspection
+        ecsFlatFieldDefinitions = (Map<String, Map<String, Object>>) ecsFlatFieldsRawMap;
+        ecsFlatMultiFieldDefinitions = new HashMap<>();
+        Iterator<Map.Entry<String, Map<String, Object>>> iterator = ecsFlatFieldDefinitions.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, Map<String, Object>> entry = iterator.next();
+            Map<String, Object> definitions = entry.getValue();
+            String type = (String) definitions.get("type");
+            if (OMIT_FIELD_TYPES.contains(type)) {
+                iterator.remove();
+            } else if ("bool".equals(type)) {
+                // for some reason, 'container.security_context.privileged' has a type "bool" in ECS definitions so we are standardizing
+                definitions.put("type", "boolean");
+            }
```

Comment:
NOTE: this is here in order to handle a single field that was [added recently](https://github.com/elastic/ecs/pull/2219/files#diff-d11ef5cbccc15e12cd888cf21575f47d4293a14f81b9fdf03f631e9cd171d391R125), probably by mistake and should be removed after this is fixed.

Reply:
Waiting for https://github.com/elastic/ecs/pull/2246 to be merged before I can remove this workaround.